cs_api_cli/cs_api_cli.ml

```diff
@@ -289,7 +293,111 @@ let run_command ~ca_file ~no_check_certificate ~api_endpoint ~api_key f =
           f ~client ~api |> Lwt_main.run))
     ~finally:(fun () -> Curl.global_cleanup ())
 
+module Ui = struct
+  let pp_level fmt (level : Logs.level) =
+    match level with
+    | Logs.App -> Fmt.styled (`Fg `White) Fmt.string fmt "APP      "
+    | Logs.Error ->
+      (Fmt.styled `Bold (Fmt.styled (`Fg `Red) Fmt.string)) fmt "ERROR    "
+    | Logs.Warning -> Fmt.styled (`Fg `Red) Fmt.string fmt "WARNING  "
+    | Logs.Info -> Fmt.styled (`Fg `Cyan) Fmt.string fmt "INFO     "
+    | Logs.Debug -> Fmt.styled (`Fg `Green) Fmt.string fmt "DEBUG    "
+
+  let format ~k ~ppf ~src:_ ~level ?prefix:_ fmt =
+    Format.kfprintf k ppf ("%a" ^^ fmt ^^ "@.") pp_level level
+
+  let reporter () =
+    let report src level ~over k msgf =
+      let k _ =
+        over ();
+        k ()
+      in
+      msgf @@ fun ?header:_ ?tags:_ fmt ->
+      format ~k ~ppf:Format.err_formatter ~src ~level fmt
+    in
+    {Logs.report}
+end
+
+let configure_logging ~level =
+  Logs.set_reporter (Ui.reporter ());
+  Logs.set_level (Some level)
+
+module Common = struct
+  module Color = struct
+    type t =
+      | Auto
+      | Always
+      | Never
+  end
+
+  module Verbosity = struct
+    type t =
+      | Quiet
+      | Normal
+      | Verbose
+  end
+
+  type t =
+    { color : Color.t
+    ; verbosity : Verbosity.t }
+
+  let common color verbosity =
+    let level =
+      match verbosity with
+      | Verbosity.Quiet -> Logs.Error
+      | Verbosity.Normal -> Logs.Info
+      | Verbosity.Verbose -> Logs.Debug
```

Comment:
**Suggestion (non-blocking):** Add a step between Normal and Quiet

I would have expected there could be a way to silence INFO messages while still having WARNING being displayed. How about making this the behavior of `Verbosity.Quiet` and adding `Verbosity.Quieter` for displaying only ERROR messages?

It could eg. be implemented by making the `-q` switch cumulative.

Reply:
Implemented your suggestion.